### PR TITLE
Port WebWorker utilities

### PR DESCRIPTION
**Pull request by chnn**
Ports over the `JobManager` and related utilities for defining `WebWorker` jobs from Chronograf.

### CODE REVIEW

File: chronograf/ui/src/workers/JobManager.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return this.workers[this.currentIndex]
+  }
+
+  private rotateWorker(): void {
+    this.currentIndex += 1
+    this.currentIndex %= workerCount
+  }
+
+  // @ts-ignore
+  private postJobRequest = async (type: string, payload) => {
```
Comment (ischolten) [inline chronograf/ui/src/workers/JobManager.ts]: could this payload be typed as `any`?

File: chronograf/ui/src/workers/worker.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import {RequestMessage, Job} from 'src/workers/types'
+
+const jobMapping: {[key: string]: Job} = {}
+
+const errorJob = async (data: RequestMessage) => {
+  postErrorResponse(data.id, new Error('Unknown job type'))
+}
+
+onmessage = async (workerMessage: {data: RequestMessage}) => {
```
Comment (ischolten) [inline chronograf/ui/src/workers/worker.ts]: should this be `onMessage`?

Comment (chnn) [reply chronograf/ui/src/workers/worker.ts]: it's this property: https://developer.mozilla.org/en-US/docs/Web/API/Worker/onmessage

😞

File: chronograf/ui/src/workers/JobManager.ts (hunk 1 above)
Comment (chnn) [reply chronograf/ui/src/workers/JobManager.ts]: sure, will change

**Pull request merged**